### ISSUE TITLE
Failure to reduce dataset with large number of y datapoints, small x span.

### ISSUE THREAD

**Issue by peterfranciscook** [tuckermcclure/matlab-plot-big#4]
Hey Tucker,

I am a big fan of this class & its methods, but it seems to not do so well for datasets with a large number of datapoints in the y span. 

I tried to workaround this using the hp = reduce_plot() wrapper, then swapping hp.XData & hp.YData, but it seemed to fail at this point, possibly because the underlying class doesn't update o.y_to_x_map when I swap the handles?

**Comment by tuckermcclure** [comment]
Indeed! This is meant for "long" datasets, like a few series of data, each with a bunch of points. For your data, can you just plot with swapped axes (that is, make your current x be y instead)? If not, then I'm not sure I understand how your data is organized; please feel free to correct me.

**Comment by peterfranciscook** [comment]
The data is from large LAS files (well logs). The "x" data is depth and the y data could be any number of things, typically a tool measurement (and typically several). The length of the file depends on several things but some of the fancier tools these days are at 6" resolution so that's 40000 * (1+ # of channels in tool array) points per file for a 20000 ft well. It is a "long" dataset in the traditional sense, but the way log data is usually presented is with depth on a reversed y axis instead of x for the semantic association of depth into the earth and the vertical span of the y axis. 

You are right -  I can use reduce plot with depth as the x axis, but then I am not presenting the data quite the way it is expected to be seen. 

I might try to fork some changes to accommodate this display.

**Comment by tuckermcclure** [comment]
Ok. Here are some things that might help.

In `get_axes_width`, you'll need to change `pixels = round(axes_position(3));` to use element 4 (the height).

You can probably leave `reduce_to_width` as it is and simply call it differently from `LinePlotReducer`, which is the file that should take most of the work.

`LinePlotReducer` will need work interpreting the inputs and then updating the data (lines 546-552). You'll need a way to specify which dimension is contains the independent variable.

If you get something working and can add it to `examples_of_line_plot_reducer` (similar to the other examples), then please open a PR, and I'll bring it in!

**Comment by hpay** [comment]
Hi Peter, 
Any luck modifying this for use with large y and small x span? I'm running into a similar situation: trying to plot 28,000 lines with only 33 points each.

**Comment by peterfranciscook** [comment]
@hpay I made some changes I haven't committed to my fork. I'll revisit the code today and push after I test it a bit more.

**Comment by peterfranciscook** [comment]
@tuckermcclure I created a pull request for the "long" vs "tall" fork. Have a look, test it out, and merge if you feel like it!

@hpay Check it out and let me know if it's what you were looking for.
